### PR TITLE
Stats: Add support Jetpack Stats banner

### PR DESCRIPTION
**Pull request by kangzj**
Related to #78653

## Proposed Changes

The PR proposes to add 'Do you love Jetpack Stats' notice, which would be shown when the site is Jetpack and the site doesn't have valid purchase (Paid). When the notice is dismissed, the notice would be snoozed for 30 days.

## Testing Instructions

* Set up Store sandbox `pejTkB-FS-p2`
* Sandbox your Jetpack
* Sandbox `public-api.wordpress.com` with D115186-code
* Build the branch for Odyssey
  * STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev
* Build Jetpack if necessary
  * jetpack build plugins/jetpack
* Open `/wp-admin/admin.php?page=stats&flags=stats/paid-stats`
* Ensure you see the notice
* Purchase Jetpack Stats `https://wordpress.com/checkout/{site-slug}/jetpack_stats_monthly`
* Ensure the notice is gone

Note: please test on Calypso Stats for a Jetpack site too

image

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-notices/do-you-love-jetpack-stats-notice.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		setTimeout( () => ( window.location.href = getStatsPurchaseURL( siteId ) ), 250 );
+	};
+
+	useEffect( () => {
+		if ( ! noticeDismissed && showNotice ) {
+			isOdysseyStats
+				? recordTracksEvent( 'jetpack_odyssey_stats_do_you_love_jetpack_stats_notice_viewed' )
+				: recordTracksEvent( 'calypso_stats_do_you_love_jetpack_stats_notice_viewed' );
+		}
+	}, [ noticeDismissed, showNotice ] );
```
Comment (dognose24) [inline client/my-sites/stats/stats-notices/do-you-love-jetpack-stats-notice.tsx]: The dependencies seem to lack `isOdysseyStats`.

截圖 2023-07-05 上午1 21 41

**Review by dognose24 — APPROVED** [review]
Works as described! 👍🏼 Left nit of lint messages.

File: client/my-sites/stats/stats-notices/do-you-love-jetpack-stats-notice.tsx (hunk 1 above)
Comment (kangzj) [reply client/my-sites/stats/stats-notices/do-you-love-jetpack-stats-notice.tsx]: Fixed in 800fb1f7dd

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8351116

Thank you @kangzj for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8351116).

**Pull request merged**